src/libsystemd-network/sd-ndisc.c

```diff
@@ -264,52 +265,84 @@ static int ndisc_recv(sd_event_source *s, int fd, uint32_t revents, void *userda
         return ndisc_handle_datagram(nd, rt);
 }
 
+static usec_t ndisc_timeout_compute_random(usec_t val) {
+        return val - val / 10 +
+                (random_u32() % (2 * USEC_PER_SEC)) * val / 10 / USEC_PER_SEC;
```

Comment:
might be nice to add an extra comment here, saying that all you do here is add a random ±10% to the value...